### DISCUSSION TITLE
How do `android_vendor_e` scripts work

### DISCUSSION THREAD

**Discussion by petefoth** [petefoth/unofficial-l4m-builds#6] [Ideas]
`android_vendor_e` has a number of script files which do interesting things.
```
finalize.sh  
init.sh  
post-sync.sh  
sync.sh
vendorsetup.sh
```
But in the `envsetup.sh, lunch, mka` build process described in the README, only `vendorsetup.sh` (and `finalize.sh` if you `mka eos`) get called. 

Are they intended for use, and can we use them in our l4m build process?

**Comment by petefoth** [comment]
Posted [question in Telegram](https://t.me/sfx_android/39819)
> @steadfasterX I'm making some progress with using android_vendor_e to make my L4M builds,  and I'm trying to get my head around how it all works.  It includes some script files which do interesting things,  but from what I can see,  in the envsetup.sh, lunch, mka build process described in the README,  only vendorsetup.sh gets called (and finalize.sh if you call `mka eos`) .
> 
> How do you use these scripts in your build process? Are they part of your automated building?

**Comment by steadfasterX** [comment]
yes all scripts are part of the automated handling. thats why only a `source build/envsetup.sh` is needed:

- `vendorsetup.sh` gets called by `source build/envsetup.sh`
- `init.sh` gets called by `vendorsetup.sh` - always (https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-q/vendorsetup.sh#L510)
- `sync.sh` gets called by `vendorsetup.sh` - when defined (https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-q/vendorsetup.sh#L549)
- `post-sync.sh` gets called by `vendorsetup.sh` - always (https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-q/vendorsetup.sh#L555)
- `finalize.sh` get called by `mka eos`

**Reply by petefoth** [reply]
>     * `post-sync.sh` gets called by `vendorsetup.sh` - always (https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-q/vendorsetup.sh#L555)

And `post-sync.sh` calls
- [`userscripts/before.sh`](https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-q/post-sync.sh#L55) **once**. This is when / how to apply the signature spoofing patches #9 
- [`userscripts/pre-build.sh`](https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-q/post-sync.sh#L55) **for each device**. 

Userscripts need to be put in .e/userscripts https://gitlab.e.foundation/steadfasterX/android_vendor_e/-/blob/v1-r/vendorsetup.sh#L39
